service/controller/chart/v1/resource/release/resource.go

```diff
@@ -89,6 +93,39 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func (r *Resource) updateAnnotations(ctx context.Context, cr v1alpha1.Chart, releaseState ReleaseState) error {
+	annotations := map[string]string{}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating annotations for app CR %#q in namespace %#q", cr.Name, cr.Namespace))
+
+	// Get chart CR again to ensure the resource version and annotations
+	// are correct.
+	currentCR, err := r.g8sClient.ApplicationV1alpha1().Charts(cr.Namespace).Get(cr.Name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	currentChecksum := key.ValuesMD5ChecksumAnnotation(*currentCR)
+
+	if releaseState.ValuesMD5Checksum != currentChecksum {
+		if currentCR.Annotations != nil {
+			annotations = currentCR.Annotations
+		}
+
+		annotations[key.ValuesMD5ChecksumAnnotationName] = releaseState.ValuesMD5Checksum
+
+		currentCR.Annotations = annotations
+		_, err = r.g8sClient.ApplicationV1alpha1().Charts(cr.Namespace).Update(currentCR)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updated annotations for app CR %#q in namespace %#q", cr.Name, cr.Namespace))
```

Comment:
This log only related to when we actually update `annotation`

Reply:
Good point. I'll fix the logging and add the missing godoc. Thx!